app/components/UserValidator/index.js

```diff
@@ -21,28 +22,31 @@ type Props = {
   searching: boolean,
 };
 
-class Validator extends Component<Props, State> {
-  input: ?HTMLInputElement;
-  state = {
-    showCompleted: false,
-  };
+const Validator = (props: Props) => {
+  const [input, setInput] = useState<?HTMLInputElement>();
+  const [completed, setCompleted] = useState(false);
+  const [showScanner, setShowScanner] = useState(false);
+  const [processing, setProcessing] = useState(false);
```

Comment:
Small thing, but is it necessary to use separate states for processing and completion?
Might be better to use `useState<'completed' | 'not_started' | 'processing'>('not_started')` or something along those lines